### PR TITLE
Remove field `blockForgeUTCTime` since `headerForgeUTCTime` is enough.

### PR DESCRIPTION
**Pull request by dnadales**
# Description

See title. Part of https://github.com/IntersectMBO/ouroboros-consensus/issues/1301

# Checklist

### Quality
* [x] Commit sequence makes sense and have useful messages, see [ref][contrib#git-history].
* [x] New tests are added and existing tests are updated.
* [x] Self-reviewed the PR.

### Maintenance
* [x] Linked an [issue][link-issue] or added the PR to the current sprint of [`ouroboros-network`][project] project.
* [x] Added labels.
* [ ] Updated changelog files.
* [x] The documentation has been properly updated, see [ref][contrib#documentation].

[project]: https://github.com/orgs/IntersectMBO/projects/5/views/1
[link-issue]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword=
[contrib#git-history]: https://github.com/IntersectMBO/ouroboros-network/blob/master/CONTRIBUTING.md#git-history
[contrib#documentation]: https://github.com/IntersectMBO/ouroboros-network/blob/master/CONTRIBUTING.md#documentation

### CODE REVIEW

File: ouroboros-network-api/CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@
      and `reRelativeStake` -> `recomputeRelativeStake`
 * Added `NodeToClientVersionV18`
 * Using `typed-protocols-0.3.0.0`.
+- Removed `blockForgeUTCTime` from `BlockFetchConsensusInterface`.
```
Comment (coot) [inline ouroboros-network-api/CHANGELOG.md:12]:
```suggestion
* Removed `blockForgeUTCTime` from `BlockFetchConsensusInterface`.
```

File: ouroboros-network/CHANGELOG.md [hunk 2]
```diff
@@ -24,6 +24,7 @@
   It is used by `outboundConnectionsState` when signaling trust state when syncing in
   Genesis mode. Default value is provided by the Configuration module.
 * Using `typed-protocols-0.2.0.0`.
+- Renamed `blockForgeUTCTime` to `headerForgeUTCTime` in `FetchClientPolicy`.
```
Comment (coot) [inline ouroboros-network/CHANGELOG.md:27]:
```suggestion
* Renamed `blockForgeUTCTime` to `headerForgeUTCTime` in `FetchClientPolicy`.
```

**Review by coot — APPROVED** [review]
LGTM; just please fix the format in `CHANGELOG`s for one that's already used.

**Comment by coot** [comment]
There's a `haskell-stylish` failure: https://ci.iog.io/build/5927288/nixlog/4

**Comment by nfrisby** [comment]
I just rebased this, handling simple conflicts with the recently merged Genesis PR.

**Pull request merged**